src/Symfony/Component/HttpKernel/Config/FileLocator.php

```diff
@@ -29,10 +29,11 @@ class FileLocator extends BaseFileLocator
      * @param string|null     $path   The path the global resource directory
      * @param array           $paths  An array of paths where to look for resources
      */
-    public function __construct(KernelInterface $kernel, string $path = null, array $paths = [])
+    public function __construct(KernelInterface $kernel, $paths = [])
     {
         $this->kernel = $kernel;
-        if (null !== $path) {
+        if (3 === \count(\func_get_args()) && \is_string(func_get_arg(1)) && null !== ($path = func_get_arg(1))) {
+            $paths = func_get_arg(2);
```

Comment:
You should trigger a deprecation here as the class is public. People can use it the old way and should be warned.

Reply:
deprecated added.